packages/expo-cli/src/urlOpts.ts

```diff
@@ -64,8 +64,14 @@ async function optsAsync(projectDir: string, options: any) {
   }
 
   let rawArgs = options.parent.rawArgs;
+
   if (hasBooleanArg(rawArgs, 'dev')) {
     opts.dev = getBooleanArg(rawArgs, 'dev');
+  } else {
+    if (rawArgs.includes('start') || rawArgs.includes('r')) {
```

Comment:
There's also `start:web` (alias `web`), which should behave in a similar way.

When it comes to the `url` and `send` commands, I think it's quite confusing that they even have the `--dev` and `--minify` options! (Even if they've existed for a long time.) Those commands are supposed to only work when you have the dev server running (started with `expo start`) and they just print/email the *manifest URL*. But I've just tested this and it turns `expo url --dev` or `expo send --minify` etc. clobber with the `dev`/`minify` settings of the project too. This won't make much sense, now that we start treating the setting as more non-persistent.

Could you remove the `--dev` and `--minify` flags from `url` and `send`? The other `urlOpts` (like `--lan` or `--tunnel`) still make sense for those commands (though I'm not sure if they should clobber the project settings either). This should allow you to get rid of these special cases for `start`/`r` and `start:web`/`web` as well, since any command that still has the `--dev` and `--minify` flags can then behave the same way.

Reply:
I removed the `http`/`https` flags as well, since if url and send are only meant for use when the server's running, then the protocol would already be configured, right?